experiments/tree_tensor_network/ttn_1d_uniform.py

```diff
@@ -903,23 +1785,175 @@ def weylops(q):
 
 
 def get_ham_potts(dtype, q, J=1.0, h=1.0):
-  U, V, om = weylops(q)
+  """Return the local term for the q-state Potts Hamiltonian.
+
+  Defines the global Hamiltonian:
+  $H = -\sum_{i=1}^N \sum_{k=1}^q [ J * U_i^k U_{i+1}^{q-k} + h * V_i^k]$
+
+  Args:
+    dtype: The data type.
+    q: Which root of unity to use. Alternatively, how many values the Potts
+      "spins" are able to take.
+    J: Coefficient for the nearest-neighbor terms (positive means
+      ferromagnetic).
+    h: Coefficient for the 1-site terms.
+
+  Returns:
+    The Hamiltonian term, separated into a 1-site contribution and a 2-site
+    MPO.
+  """
+  U, V, _ = _weylops(q)
 
   mp = np.linalg.matrix_power
 
-  h2 = [-J * mp(U, k) for k in range(1, q)], [mp(U, q - k) for k in range(1, q)]
+  if dtype_is_complex(dtype):
+    h2 = ([-J * mp(U, k) for k in range(1, q)],
+          [mp(U, q - k) for k in range(1, q)])
+  else:
+    # The straightforward way to build the Hamiltonian results in complex
+    # matrices in the MPO. The dense Hamiltonian is, however, real.
+    # To make the MPO real, we first build the dense 2-site term, then
+    # use an SVD to split it back into a real MPO.
+    h2_dense = sum(np.tensordot(-J * mp(U, k), mp(U, q-k), axes=((),()))
+                for k in range(1,q))
+    realness = np.linalg.norm(h2_dense - h2_dense.real)
+    if realness > 1e-12:
+      raise ValueError(
+        "2-site term was not real. Realness = {}".format(realness))
+    u, s, vh = svd_np(h2_dense.real.reshape((q**2, q**2)), full_matrices=False)
+    mpo_rank = np.count_nonzero(s.round(decimals=12))
+    if mpo_rank != q - 1:
+      raise ValueError(
+        "Error performing SVD of 2-site term. {} != {}-1".format(mpo_rank, q))
+    h2 = ([s[i] * u[:,i].reshape(q,q) for i in range(q-1)],
+          [vh[i,:].reshape(q,q) for i in range(q-1)])
+
   h1 = -h * sum(mp(V, k) for k in range(1, q))
 
-  h1 = tf.convert_to_tensor(h1, dtype=dtype)
+  h1 = convert_to_tensor(h1, dtype=dtype)
   h2 = (
-      [tf.convert_to_tensor(h, dtype=dtype) for h in h2[0]],
-      [tf.convert_to_tensor(h, dtype=dtype) for h in h2[1]],
+      [convert_to_tensor(h, dtype=dtype) for h in h2[0]],
+      [convert_to_tensor(h, dtype=dtype) for h in h2[1]],
   )
 
   return h1, h2
 
 
+def get_ham_heis_su3_2box(dtype):
+  """Return the local term for the su(3) Heisenberg model at 2-box level.
+
+  VERY EXPERIMENTAL: Requires su3 irrep matrices!
+
+  Args:
+    dtype: The data type.
+
+  Returns:
+    The Hamiltonian term, separated into a 1-site contribution and a 2-site
+    MPO.
+  """
+  import scipy.io as sio
+  su3_20 = sio.loadmat("experiments/tree_tensor_network/su3_20.mat")
```

Comment:
Don't hard code paths. If eventually want to use this outside of experimental this will break.

Reply:
Yeah. Actually, this file is not even included in the PR. Probably I should just take this function out for now...